ldap.sgnl.yaml

```
@@ -0,0 +1,151 @@
+description: "LDAP as a System of Record"
+hostname: "{{Input Required: test.data.com}}"
+defaultSyncFrequency: HOURLY
+defaultSyncMinInterval: 1
+defaultApiCallFrequency: SECONDLY
+defaultApiCallMinInterval: 1
+type: "LDAP-1.0.0"
+# Example Config:
+# {
+#   "baseDN":"dc=org,dc=example,dc=io",
+#   "certificateChain": ". . . base64 encoded certificate chain here . . "
+#   "entityConfig": {
+#     "<EntityName-1>": {
+#       "query": "(&(objectCategory=user)(objectClass=user)(distinguishedName=*))"
+#     },
+#     "<EntityName-2>": {
+#       "query": "(&(objectCategory=group)(objectClass=group)(distinguishedName=*))"
+#     },
+#     "<EntityName-3>": {
+#       "memberOf": <EntityName-2>,
+#       "query": "(&(memberOf={{CollectionId}})(objectCategory=user)(objectClass=user))",
+#       "memberUniqueIdAttribute": "memberObjectGUID",
+#       "memberOfUniqueIdAttribute": "groupObjectGUID"
+#     }
+#   },
+# }
+
+# Adapter Config Field Info:
+# [Required] baseDN (String): Specifies the Base Distinguished Name (DN) for the LDAP directory.
+
+# [Required] certificateChain (String): Contains the Base64 encoded certificate chain for secure communication with the LDAP server.
+
+# [Optional] entityConfig (Object): Configurations for different types of entities within the LDAP directory.
+
+# For more information on LDAP queries: https://ldap.com/ldap-filters/
+
+# [Optional] entityConfig.<EntityName> (Object): Configuration for a specific type of entity.
+# [Optional] entityConfig.<EntityName>.query (String): Specifies the LDAP query for retrieving entities of this type.
+
+# [Optional] entityConfig.<EntityName>.memberOf (String): Specifies the group or entity to which the members belong.
+# [Optional] entityConfig.<EntityName>.query (String): Specifies the LDAP query for retrieving entities of this type.
+# [Optional] entityConfig.<EntityName>.memberUniqueIdAttribute (String): Specifies the attribute used as a unique identifier for members.
+# [Optional] entityConfig.<EntityName>.memberOfUniqueIdAttribute (String): Specifies the attribute used as a unique identifier for the group or parent entity.
+
+
+adapterConfig: "ewogICAgImJhc2VETiI6ICJ7e0lucHV0IFJlcXVpcmVkOiBCYXNlRE59fSIsCiAgICAiY2VydGlmaWNhdGVDaGFpbiI6ICJ7e0lucHV0IFJlcXVpcmVkOiBJZiBMREFQUyBpcyBlbmFibGVkOiBiYXNlNjQgZW5jb2RlZCBjZXJ0aWZpY2F0ZUNoYWlufX0iCn0="
```

Placeholder comment to update the config based on the example entities chosen.
```suggestion
adapterConfig: "ewogICAgImJhc2VETiI6ICJ7e0lucHV0IFJlcXVpcmVkOiBCYXNlRE59fSIsCiAgICAiY2VydGlmaWNhdGVDaGFpbiI6ICJ7e0lucHV0IFJlcXVpcmVkOiBJZiBMREFQUyBpcyBlbmFibGVkOiBiYXNlNjQgZW5jb2RlZCBjZXJ0aWZpY2F0ZUNoYWlufX0iCn0="
```